src/h5web/vis-packs/core/utils.ts

```diff
@@ -181,22 +167,29 @@ export function extendDomain(
   }
 
   const [min, max] = domain;
-  const isLog = scaleType === ScaleType.Log;
+  const { supportedDomain, createScale } = H5WEB_SCALES[scaleType];
+  const scale = createScale({ domain, range: [0, 1] });
+  const extendedDomain: Domain = [
+    clampBound(scale.invert(-extendFactor), supportedDomain),
+    clampBound(scale.invert(1 + extendFactor), supportedDomain),
+  ];
 
-  if (min <= 0 && isLog) {
-    throw new Error('Expected domain compatible with log scale');
+  const h5webScale = H5WEB_SCALES[scaleType];
+
+  if (!h5webScale.supportsDomain(domain)) {
```

Comment:
This function bothers me. For some scales, we have a `supportedDomain` and yet `supportsDomain: () => true`. This feels incoherent.

Also, I feel like this refactoring of `extendDomain` changes the behaviour of the function significantly: you create the scale and call it before checking if the scale is compatible. Not to mention that `min * max > 0` is not the same as checking `min <= 0`. Are you sure this refactoring doesn't have unintended consequences?